pkg/services/sqlstore/database_config.go

```diff
@@ -235,3 +235,33 @@ func buildExtraConnectionString(sep rune, urlQueryParams map[string][]string) st
 	}
 	return sb.String()
 }
+
+func validateReplicaConfigs(primary *DatabaseConfig, cfgs []*DatabaseConfig) error {
+	if cfgs == nil {
+		return errors.New("cfg cannot be nil")
+	}
+
+	// Return multiple errors so we can fix them all at once!
+	var result error
+
+	// Check for duplicate connection strings
+	seen := make(map[string]struct{})
+	seen[primary.ConnectionString] = struct{}{}
+	for _, cfg := range cfgs {
+		if _, ok := seen[cfg.ConnectionString]; ok {
+			result = errors.Join(result, fmt.Errorf("found duplicate connection string: %s", cfg.ConnectionString))
```

Comment:
Would it be possible that `cfg.ConnectionString` contains a secret?

Reply:
It CAN and almost always DOES and that's why I haven't been recording the actual string anywhere else, so I am very grateful you caught this :D